src/formatting_stack/branch_formatter.clj

```diff
@@ -75,37 +81,60 @@
                                 strategies/namespaces-within-refresh-dirs-only)))])
 
-(def default-processors
+(speced/defn default-processors [_
+                                 ^vector? processors-strategies
+                                 ^map? third-party-indent-specs]
   [(processors.cider/new {:third-party-indent-specs third-party-indent-specs})])
 
 (def default-reporter
   (pretty-printer/new {}))
 
-(defn format-and-lint-branch! [& {:keys [target-branch in-background? reporter]
-                                  :or   {target-branch  "master"
-                                         in-background? (not (System/getenv "CI"))
-                                         reporter       default-reporter}}]
+(def formatter-factory
+  (implement {}
+    protocols.formatter/--formatters default-formatters))
+
+(def no-formatters (constantly []))
+
+(def empty-formatter-factory
+  (implement {}
+    protocols.formatter/--formatters no-formatters))
+
+(def linter-factory
+  (implement {}
+    protocols.linter/--linters default-linters))
+
+(def processor-factory
+  (implement {}
+    protocols.processor/--processors default-processors))
+
+(defn format-and-lint-branch!
+  "Note that files that are not completely staged will not be affected.
+
+  You can override that with `:blacklist []`."
+  [& {:keys [target-branch in-background? reporter formatters blacklist overrides]
+      :or   {target-branch  "master"
+             in-background? (not (System/getenv "CI"))
+             reporter       default-reporter}}]
   (let [default-strategies [(fn [& {:as options}]
-                              (mapply strategies/git-diff-against-default-branch (assoc options :target-branch target-branch)))]
-        formatters (default-formatters default-strategies)
-        linters (default-linters default-strategies)]
+                              (mapply strategies/git-diff-against-default-branch (cond-> options
+                                                                                   true      (assoc :target-branch target-branch)
+                                                                                   blacklist (assoc :blacklist blacklist))))]]
     (formatting-stack.core/format! :strategies default-strategies
-                                   :processors default-processors
-                                   :formatters formatters
+                                   :overrides overrides
+                                   :formatters (or formatters formatter-factory)
+                                   :linters linter-factory
+                                   :processors processor-factory
                                    :reporter reporter
-                                   :linters linters
                                    :in-background? in-background?)))
 
-(defn lint-branch! [& {:keys [target-branch in-background? reporter]
+(defn lint-branch! [& {:keys [target-branch in-background? reporter overrides]
```

Comment:
Why nest formatters/linters/processors/strategies under `override`? They seem direct arguments for `lint-branch!`

Reply:
The first thing that comes to mind is that the config should be a single map, so that it can become _data_.

* data (a single hashmap) is simpler than code (a more intrincate invocation consisting of multiple arguments)
* a single hashmap can be (more) easily a resource, a piece of Lein config, etc.
  * this is important, as one will want to share config across projects.
    * I envision this being accomplished by creating a single dep, containing  a .edn as a resource. Zero config or `require`s to be done for consumers: we just _discover_ the presence of such a config file.
* Hashmaps can be (deep) merged
  * Not hard to imagine that people can want a merge of f-s defaults <-> workplace defaults <-> personal defaults.

It's also good to emphasize the notion of overrides as:

* a thing you should be aware of
  * if I can pass `:linters` separately, I can be unaware of the existence a deep-merged overrides system
  * unawareness -> confusion, surprises, etc
* the sole intended API
  * More APIs -> more complexity (surface area for bugs)